File: service/src/main/java/hu/bsstudio/robonaut/team/model/DetailedTeam.java
package hu.bsstudio.robonaut.team.model;

import hu.bsstudio.robonaut.entity.TeamType;
import java.util.List;
import lombok.Builder;
import lombok.Value;
import lombok.extern.jackson.Jacksonized;

@Value
@Builder
@Jacksonized
public class DetailedTeam {
    long teamId;
    int year;
    String teamName;
    List<String> teamMembers;
    TeamType teamType;

    int skillScore;

    int numberOfOvertakes;
    boolean safetyCarWasFollowed;

    List<Integer> speedTimes;

    int votes;
    int audienceScore;

    int qualificationScore;

    Score combinedScore;
    Score juniorScore;
}

File: service/src/main/kotlin/hu/bsstudio/robonaut/team/model/DetailedTeam.kt
package hu.bsstudio.robonaut.team.model

import hu.bsstudio.robonaut.entity.TeamType

data class DetailedTeam (
    val teamId: Long,
    val year: Int,
    val teamName: String,
    val teamMembers: List<String>,
    val teamType: TeamType,
    val skillScore: Int,
    val numberOfOvertakes: Int,
    val safetyCarWasFollowed: Boolean,
    val speedTimes: List<Int>,
    val votes: Int,
    val audienceScore: Int,
    val qualificationScore: Int,
    val combinedScore: Score,
    val juniorScore: Score,
)
